python_modules/dagster/dagster/daemon/sensor.py

```diff
@@ -359,8 +359,46 @@ def execute_sensor_iteration(
         elif _is_under_min_interval(sensor_state, external_sensor, now):
             continue
 
-        future = executor.submit(
-            _process_tick,
+        if isinstance(executor, SynchronousExecutor):
+            yield from _process_tick_generator(
+                logger,
+                instance,
+                workspace,
+                now,
+                external_sensor,
+                sensor_state,
+                sensor_debug_crash_flags,
+                tick_retention_settings,
+            )
+        else:
+            future = executor.submit(
+                _process_tick,
+                logger,
+                instance,
+                workspace,
+                now,
+                external_sensor,
+                sensor_state,
+                sensor_debug_crash_flags,
+                tick_retention_settings,
+            )
+            yield
+            if debug_futures is not None:
+                debug_futures[external_sensor.selector_id] = future
```

Comment:
as per DMs - with this "unified interface" breaking down just delete sync executor and embrace the fork assuming thats reasonable